doc/classes/NavigationAgent.xml

```diff
@@ -0,0 +1,163 @@
+			</description>
+		</method>
+		<method name="get_final_location">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns the reachable final location in global coordinates. This can change if the navigation path is altered in any way.
+			</description>
+		</method>
+		<method name="get_nav_path" qualifiers="const">
+			<return type="PoolVector3Array">
+			</return>
+			<description>
+				Returns the path from start to finish in global coordinates.
+			</description>
+		</method>
+		<method name="get_nav_path_index" qualifiers="const">
+			<return type="int">
+			</return>
+			<description>
+				Returns which index the agent is currently on in the navigation path's [PoolVector3Array].
+			</description>
+		</method>
+		<method name="get_navigation" qualifiers="const">
+			<return type="Node">
+			</return>
+			<description>
+				Returns the [Navigation] node that the agent is using for its navigation system.
+			</description>
+		</method>
+		<method name="get_next_location">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns a [Vector3] in global coordinates, that can be moved to, making sure that there are no static objects in the way.
+			</description>
+		</method>
+		<method name="get_target_location" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns the user defined [Vector3] after setting the target location.
+			</description>
+		</method>
+		<method name="is_navigation_finished">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the navigation path's final location has been reached.
+			</description>
+		</method>
+		<method name="is_target_reachable">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the target location is reachable. The target location is set using [method set_target_location].
+			</description>
+		</method>
+		<method name="is_target_reached" qualifiers="const">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the target location is reached. The target location is set using [method set_target_location]. It may not always be possible to reach the target location. It should always be possible to reach the final location though. See [method get_final_location].
+			</description>
+		</method>
+		<method name="set_navigation">
+			<return type="void">
+			</return>
+            <argument index="0" name="navigation" type="Node">
+            </argument>
+			<description>
+				Sets the [Navigation] node used by the agent. Useful when you don't want to make the agent a child of a [Navigation] node.
+			</description>
+		</method>
+		<method name="set_target_location">
+			<return type="void">
+			</return>
+            <argument index="0" name="location" type="Vector3">
+            </argument>
+			<description>
+				Sets the user desired final location. This will clear the current navigation path.
```

Comment:
It's better to not tell about the path because the idea is that the user have to care about only the next location

Reply:
I don't think it is a good idea to obfuscate any part of it. I believe users SHOULD know that if they call this, it does clear the current navigation path. While it may be a small detail in a vast majority of the cases, it may save someone a headache. Even if it is just one person.